golem-common/src/model/public_oplog.rs

```diff
@@ -328,6 +330,396 @@ pub enum PublicOplogEntry {
     Restart(TimestampParameter),
 }
 
+impl PublicOplogEntry {
+    pub fn matches(&self, query: &Query) -> bool {
+        fn matches_impl(entry: &PublicOplogEntry, query: &Query, field_stack: &[String]) -> bool {
+            match query {
+                Query::Or { queries } => queries
+                    .iter()
+                    .any(|query| matches_impl(entry, query, field_stack)),
+                Query::And { queries } => queries
+                    .iter()
+                    .all(|query| matches_impl(entry, query, field_stack)),
+                Query::Not { query } => !matches_impl(entry, query, field_stack),
+                Query::Regex { .. } => {
+                    entry.matches_leaf_query(field_stack, &query.clone().try_into().unwrap())
```

Comment:
Why is this unwrap?

Reply:
`Regex` is a leaf node so that `try_into` always succeeds.